server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -712,7 +715,8 @@ private IndexMetadata(
         this.priority = priority;
         this.creationDate = creationDate;
         this.ignoreDiskWatermarks = ignoreDiskWatermarks;
-        this.tierPreference = tierPreference;
+        // always configure the correct (one and only) value for the partially mounted indices tier preference
+        this.tierPreference = isPartialSearchableSnapshot ? PARTIALLY_MOUNTED_INDEX_TIER_PREFERENCE : tierPreference;
```

Comment:
@original-brownbear as discussed, we're only interpreting the tier preference here and not changing the underlying setting